plugins/aws-proxy-filebased-credentials-provider/src/main/java/io/trino/aws/proxy/plugins/filebasedcredentialsprovider/FileBasedCredentialsProviderPlugin.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.plugins.filebasedcredentialsprovider;
+
+import com.google.inject.Module;
+import io.trino.aws.proxy.spi.TrinoAwsProxyServerPlugin;
+
+public class FileBasedCredentialsProviderPlugin
```

Comment:
`TrinoAwsProxyServerPlugin` are loaded via the JDK Service Loader. You need to add the META-INF services file.

Reply:
Trino "plugins" are connectors loaded in their own isolated classloader sandboxes. They are complex, nearly complete applications with their own Bootstrap, etc. We don't need something that complex for this. The Trino AWS Proxy will be a set of JARs that serve as the building blocks for a single use-case proxy. Users of the Trino AWS Proxy will mixin their own credentials providers, etc. These mixin/plugins shouldn't be isolated sandboxes. They are part of the main application and, thus, the simpler (standard JDK) plugin mechanism is much better in this instance.

Reply:
In terms of reading the file via config - that's fine. Let's leave that.

Reply:
Moved to a package